cypress/integration/signin_authentication/forgot_password_spec.js

```diff
@@ -0,0 +1,146 @@
+// ***************************************************************
+
+import users from '../../fixtures/users.json';
+
+import {reUrl} from '../../utils';
+
+const user1 = users['user-1'];
+
+const feedbackEmail = 'test@example.com';
+
+describe('Signin/Authentication', () => {
+    before(() => {
+        cy.apiUpdateConfig({EmailSettings: {FeedbackEmail: feedbackEmail}});
+    });
+
+    it('SA15008 - Sign In Forgot password - Email address has account on server', () => {
+        resetPasswordAndLogin(user1, feedbackEmail, false);
+    });
+
+    it('Revert original password of user-1', () => {
+        resetPasswordAndLogin(user1, feedbackEmail, true);
+    });
+});
+
+function verifyForgotPasswordEmail(response, toUser, fromEmail) {
+    const isoDate = new Date().toISOString().substring(0, 10);
+    const {data, status} = response;
+
+    // * Should return success status
+    expect(status).to.equal(200);
+
+    // * Verify that email is addressed to user-1
+    expect(data.to.length).to.equal(1);
+    expect(data.to[0]).to.contain(toUser.email);
+
+    // * Verify that email is from default feedback email
+    expect(data.from).to.contain(fromEmail);
+
+    // * Verify that date is current
+    expect(data.date).to.contain(isoDate);
+
+    // * Verify that the email subject is correct
+    expect(data.subject).to.equal('[Mattermost] Reset your password');
+
+    // * Verify that the email body is correct
+    const bodyText = data.body.text.split('\r\n');
+    expect(bodyText.length).to.equal(14);
+    expect(bodyText[1]).to.equal('You requested a password reset');
+    expect(bodyText[4]).to.equal('To change your password, click "Reset Password" below.');
+    expect(bodyText[5]).to.contain('If you did not mean to reset your password, please ignore this email and your password will remain the same. The password reset link expires in 24 hours.');
+    expect(bodyText[7]).to.contain('Reset Password');
+    expect(bodyText[9]).to.contain('Any questions at all, mail us any time: feedback@mattermost.com');
+    expect(bodyText[10]).to.equal('Best wishes,');
+    expect(bodyText[11]).to.equal('The Mattermost Team');
+    expect(bodyText[13]).to.equal('To change your notification preferences, log in to your team site and go to Account Settings > Notifications.');
+}
+
+function resetPasswordAndLogin(user, fromEmail, revertWithEnterKey) {
+    const newPassword = 'newpasswd';
+
+    // # Visit '/'
+    cy.visit('/');
+
+    // * Verify that it redirects to /login
+    cy.url().should('contain', '/login');
+
+    // * Verify that forgot password link is present
+    // # Click forgot password link
+    cy.get('#login_forgot > a').should('be.visible').and('have.text', 'I forgot my password').click();
+
+    // * Verify that it redirects to /reset_password
+    cy.url().should('contain', '/reset_password');
+
+    // * Verify that the focus is set to passwordResetEmailInput
+    cy.focused().should('have.attr', 'id', 'passwordResetEmailInput');
```

Comment:
In 3.3.0 (https://docs.cypress.io/guides/references/changelog.html#3-3-0) they added have.focus, not.have.focus, be.focused and not.be.focused assertions, once we update we should update these assertions, there aren't too many where we are doing things with focused elements. Obviously, no change needed now.

Reply:
Yup, noted, let's do that.  How about change at the same time when upgrading Cypress?

Reply:
> they added have.focus, not.have.focus, be.focused and not.be.focused assertions,

Good to know 👍